src/IIIFPresentation/API/Features/Manifest/CanvasPaintingMerger.cs

```diff
@@ -76,7 +76,9 @@ private static void CheckForDuplicates(List<InterimCanvasPainting> itemsCanvasPa
         if (matchedCanvasPainting.Count != 0)
         {
             throw new CanvasPaintingMergerException(
-                $"Canvas painting records with the following id's conflict with the order from items - {string.Join(',', matchedCanvasPainting.Select(m => m.Id))}");
+                "The following canvas painting records conflict with the order from items - " +
+                $"{string.Join(',', 
+                    matchedCanvasPainting.Select(m => $"{{{(m.Id != null ? $"id: {m.Id}, " : "")}canvasOrder: {m.CanvasOrder}}}"))}");
```

Comment:
Can this be cleaned up at all, this one line ends `}}}"))}")` 😕 

There's a warning for `m.Id != null` - can that be fixed (`string.IsNullOrEmpty()` maybe)? _Can_ it be null - if so should the type be nullable?

> Expression is always true according to nullable reference types' annotations